src/Native/System.Native/pal_networking.cpp

```diff
@@ -392,3 +360,270 @@ extern "C" int32_t GetHostName(uint8_t* name, int32_t nameLength)
     size_t unsignedSize = UnsignedCast(nameLength);
     return gethostname(reinterpret_cast<char*>(name), unsignedSize);
 }
+
+template <typename TType, typename TField>
+static bool IsInBounds(const TType* base, size_t len, const TField* value)
+{
+    auto* baseAddr = reinterpret_cast<const uint8_t*>(base);
+    auto* valueAddr = reinterpret_cast<const uint8_t*>(value);
+    return valueAddr >= baseAddr && (valueAddr + sizeof(TField)) <= (baseAddr + len);
+}
+
+extern "C" Error GetIPSocketAddressSizes(int32_t* ipv4SocketAddressSize, int32_t* ipv6SocketAddressSize)
+{
+    if (ipv4SocketAddressSize == nullptr || ipv6SocketAddressSize == nullptr)
+    {
+        return PAL_EFAULT;
+    }
+
+    *ipv4SocketAddressSize = sizeof(sockaddr_in);
+    *ipv6SocketAddressSize = sizeof(sockaddr_in6);
+    return PAL_SUCCESS;
+}
+
+extern "C" Error GetAddressFamily(const uint8_t* socketAddress, int32_t socketAddressLen, int32_t* addressFamily)
+{
+    if (socketAddress == nullptr || addressFamily == nullptr || socketAddressLen < 0)
+    {
+        return PAL_EFAULT;
+    }
+
+    auto* sockAddr = reinterpret_cast<const sockaddr*>(socketAddress);
+    if (!IsInBounds(sockAddr, static_cast<size_t>(socketAddressLen), &sockAddr->sa_family))
+    {
+        return PAL_EFAULT;
+    }
+
+    switch (sockAddr->sa_family)
+    {
+        case AF_UNSPEC:
+            *addressFamily = PAL_AF_UNSPEC;
+            return PAL_SUCCESS;
+
+        case AF_UNIX:
+            *addressFamily = PAL_AF_UNIX;
+            return PAL_SUCCESS;
```

Comment:
With this, do Socket's "just work" with domain sockets, or is there work necessary elsewhere in the pal to make them work?  If they just work, great!  If not, I'm wondering whether it'd be better to remove this case for now and just have PAL_EAFNOSUPPORT returned until such time that appropriate changes are made across the layer to enable them.